seacas: require +metis and +mpi instead of +parmetis

mvlopri opened the pull request:
This change aligns the build condition for parmetis with the depends_on condition.
The current build condition of parmetis looks for "+parmetis" in the spec which is not added by the depends_on as that adds "^parmetis" instead.

<!--  
Remember that `spackbot` can help with your PR in multiple ways:
- `@spackbot help` shows all the commands that are currently available
- `@spackbot fix style` tries to push a commit to fix style issues in this PR
- `@spackbot re-run pipeline` runs the pipelines again, if you have write access to the repository 
-->

davi0011 committed:
Change build condition of parmetis to +metis and +mpi

This change aligns the build condition for parmetis with the
depends_on condition.
The current build condition of parmetis looks for "+parmetis" in
the spec which is not added by the depends_on as that adds
"^parmetis" instead.

mvlopri commented:
@spackbot fixstyle

mvlopri commented:
@spackbot commands

mvlopri commented:
@spackbot fix style

alalazo committed:
Remove "prevent_cray_detection" context manager (#46060)

This context manager was used to prevent detecting a platform
as "cray". Since now Cray machines are detected as linux, we can
remove the context manager.

haampie committed:
spack_yaml: delete custom deepcopy (#46048)

ye-luo committed:
Introduce offload variant for llvm >= 19. (#45865)

Try2Code committed:
cdo-2.4.3 (#46033)

cdo-2.4.3 - fixed hash

adamjstewart committed:
py-numpy: add v2.0.2 (#46056)

alalazo committed:
Remove llnl.util.lang.has_method, use built-in hasattr instead (#46072)

alalazo committed:
Spec.from_detection now accounts for external prefix (#46063)

Change the signature of Spec.from_detection to set the
external prefix, and the external modules, if they are
present.

Delete "spack.package_prefs.spec_externals" since it
is unused.

mabraham committed:
gromacs: add conflict between NVSHMEM and cuFFTMp (#46065)

* Add conflict between NVSHMEM and cuFFTMp for GROMACS package

These don't work in the same build configuration.

* [@spackbot] updating style on behalf of mabraham

* Update package.py

Also constrain NVSHMEM appropriately

* Update var/spack/repos/builtin/packages/gromacs/package.py

Co-authored-by: Andrey Alekseenko <al42and@gmail.com>

* Update package.py

* [@spackbot] updating style on behalf of mabraham

* Update package.py

* Update package.py

* Update package.py

* [@spackbot] updating style on behalf of mabraham

---------

Co-authored-by: mabraham <mabraham@users.noreply.github.com>
Co-authored-by: Christoph Junghans <christoph.junghans@gmail.com>
Co-authored-by: Andrey Alekseenko <al42and@gmail.com>

haampie committed:
Drop now redundant use of inspect (#46057)

inspect.isclass -> isinstance(..., type)
inspect.getmro -> cls.__mro__

haampie committed:
buildcache.py: elide string not spec (#46074)

msimberg committed:
boost: install `BoostConfig.cmake` even when header-only (#46062)

* Install BoostConfig.cmake even when header-only

* boost: Only use --without-libraries when --with-libraries would have an empty list

nicholas-sly committed:
Replace unparsable apostrophe character with ASCII "'". (#46069)

Jordan474 committed:
For "when:" and install_environment.json: Support fully qualified hostname (#45522)

balay committed:
mfem, pflotran, alquimia: remove old versions with xsdk string (in version) that were used in old/removed xsdk releases (#45837)

Jordan474 committed:
Don't check checksums on spack-develop packages (#46076)

Fix regression introduced in spack 0.22.1 where Spack would ask about checksums
on spack-develop packages.

robertu94 committed:
py-numcodecs rename git branch to match upstream (#46085)

* py-numcodecs rename git branch to match upstream

renjithravindrankannath committed:
llvm-amdgpu: Updating LD_LIBRARY_PATH w.r.t new prefix path (#45940)

* Updating LD_LIBRARY_PATH w.r.t new prefix path

* Updating hsa external path for 6.x

drkrynstrng committed:
pmix: add v5.0.3 and fix variants (#45621)

* Add pmix v5.0.3 and fix variants
* Update pmix homepage link
* pmix: Simplify/update hwloc dependency
* pmix: Update versions for --disable-sphinx configure option
* pmix: Add munge variant
* pmix: Add zlib dependency
* pmix: Fix dependency py-sphinx@5

mtaillefumier committed:
[Update] New version of sirius (#46049)

EbiArnie committed:
perl-compress-bzip2: new package (#46055)

* perl-compress-bzip2: new package
  Adds Compress::Bzip2
* Use bzip2 from Spack, not system

gsjaardema reviewed (approved):
This looks good, but do the references to parmetis in that block need to change also:
```
                    define(
                        "ParMETIS_LIBRARY_DIRS",
                        [spec["parmetis"].prefix.lib, spec["metis"].prefix.lib],
                    ),
                    define("ParMETIS_LIBRARY_NAMES", ["parmetis", "metis"]),
                    define(
                        "TPL_ParMETIS_INCLUDE_DIRS",
                        [spec["parmetis"].prefix.include, spec["metis"].prefix.include],
                    ),
```
In particular the `spec["parmetis"]`.  They might be OK, but just asking to be sure.

etiennemlb committed:
Add more compiler leniency (#46083)

alalazo committed:
Remove "get_compiler_duplicates" (#46077)

This function is used only in tests.

haampie committed:
tau: fix (cray) compiler names/paths (#46104)

fixes a build issue on cray ci

tldahlgren committed:
Executable: make the timeout message readable (#46098)

wdconinc committed:
py-mypy: add v1.11.2 (#46099)

albestro committed:
paraview: add cdi support (#44222)

* add basic CDI package

* add CDI variant to paraview

* [@spackbot] updating style on behalf of albestro

---------

Co-authored-by: albestro <albestro@users.noreply.github.com>

haampie committed:
windows_establish_runtime_linkage: post install hook (#46103)

haampie committed:
py-greenlet: add missing forward compat bound (#46113)

haampie committed:
spectrum-mpi: no windows (#46119)

louspe-linaro committed:
linaro-forge: added 24.0.4 version (#46112)

Updated platform.machine() to only match "aarch64".
"arm64" not supported by "spack arch"

Chrismarsh committed:
CMakePackage: Set policy CMP0042 NEW on macos (#46114)

so linking to shared libraries works well at runtime on darwin for all packages with cmake_minimum_required < 3.

white238 committed:
conduit package: Honor compiler extra_rpaths + extras (#46046)

- Honor compiler extra_rpaths (this build bypasses spack wrappers,
  so the RPATHs are added explicitly as CMake options)
- Use root_cmakelists_dir instead of adding it to the command line
- Add BLT as a dependency, allowing different versions outside of what
  is in the tarball
- Put a copy of host-config in the stage directory: this allows
  examination of the host-config when a build fails (before, the
  host-config was just stored in the install directory, which is
  deleted by default on a failed build)

mabraham committed:
gromacs: support version 2024.3 (#46117)

haampie committed:
cray: allow failure due to broken blas (#46111)

1. libsci_cray.so is broken, as it fails to list `libdl.so` in
   DT_NEEDED.
2. cray compilers fail to build a different blas

alalazo committed:
mpitrampoline: fix wrong use of compiler.cc_names (#46038)

haampie committed:
windows-vis: vtk~mpi (#46133)

alalazo committed:
Remove `DetectedPackage` class (#46071)

This PR simplifies the code doing external spec detection by removing 
the `DetectedPackage` class. Now, functions accepting or returning lists 
of `DetectedPackage`, will accept or return list of specs.

Performance doesn't seem to change if we use `Spec.__reduce__` instead 
of `DetectionPackage.__reduce__`.

alalazo committed:
Put some more constraint on a few mpi providers (#46132)

This should help not selecting, by default, some niche implementation that are supposed to be externals.

Signed-off-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

haampie committed:
windows-vis: fix failing pipeline (#46135)

* seacas: fix gnu parallel dep

* add vtk@9.0 platform=windows conflict

alalazo committed:
boost: remove Compiler.cxx_names (#46037)

draenog committed:
plumed: add v2.9.1  (#46022)

etiennemlb committed:
unzip: use more generic strip flag for cce (#46087)

* Use more generic strip flag for cce

* [@spackbot] updating style on behalf of etiennemlb

* Apply always

dependabot[bot] committed:
build(deps): bump actions/upload-artifact from 4.3.6 to 4.4.0 (#46149)

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 4.3.6 to 4.4.0.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/834a144ee995460fba8ed112a2fc961b36a5ec5a...50769540e7f4bd5e21e526ee35c689e35e0d6874)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

AlexanderRichert-NOAA committed:
sfcio: add v1.4.2 (#46146)

* sfcio: add v1.4.2

* [@spackbot] updating style on behalf of AlexanderRichert-NOAA

AlexanderRichert-NOAA committed:
gfsio: add v1.4.2 (#46145)

AlexanderRichert-NOAA committed:
landsfcutil: add v2.4.2 (#46144)

pranav-sivaraman committed:
doctest: add new package (#46138)

dependabot[bot] committed:
build(deps): bump actions/setup-python from 5.1.1 to 5.2.0 (#46129)

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 5.1.1 to 5.2.0.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/39cd14951b08e74b54015e9e001cdefcf80e669f...f677139bbe7f9c59b41e40162b753c062f5d49a3)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

AlexanderRichert-NOAA committed:
sigio: add v2.3.3 (#46116)

qwertos committed:
New package: py-ops (#46122)

* New package: py-ops

* [py-ops]

- added version 2.16.0
- ran black
- updated copyright
- added license()

---------

Co-authored-by: vehrc <vehrc@rit.edu>

qwertos committed:
New package: py-jsonlines (#46124)

* py-jsonlines: new package

* py-jsonlines: fix dependency

---------

Co-authored-by: Alex C Leute <acl2809@rit.edu>

stephenswat committed:
podio: apply patch for gcc 14 builds (#45854)

* podio: apply patch for gcc 14 builds

Podio versions after 0.17.0 but before 1.0.0 are broken when using gcc
14 because of a missing include, which is addressed in the podio pull
request at https://github.com/AIDASoft/podio/pull/600. This commit
patches pre-1.0.0 versions of Podio so they can be compiled with gcc 14,
which is important for building Acts.

* Style

* Style 2

* Fixes

* Add comment:

* Add sha256

jmcarcell committed:
whizard: add a patch when using hepmc3 3.3.0 or newer (#45862)

* whizard: add a patch when using hepmc3 3.3.0 or newer

* whizard: comment with patch origin

---------

Co-authored-by: jmcarcell <jmcarcell@users.noreply.github.com>
Co-authored-by: Wouter Deconinck <wdconinc@gmail.com>

mpokorny committed:
casacore: add new versions 3.6.1, 3.6.0, 3.2.1 (#46068)

dunatotatos committed:
Update py-numcodecs. (#45715)

haampie committed:
Remove redundant inspect.getmodule(self) idiom in packages (#46073)

joequant committed:
add rapidjson conflict for gcc14 (#46007)

stephenswat committed:
dd4hep: mark conflict with root@6.31.1: (#45855)

dd4hep versions up to and including 1.27 had a conflict with root
versions starting from 6.31.1, as shown in
https://github.com/AIDASoft/DD4hep/issues/1210. This PR explicitly adds
that conflict to the spec.

Commit:
[openmpi] Add optional debug build variant (#45708)

Pandapip1 committed:
nanomath: add version 1.4.0 (#46159)

prudhomm committed:
package cln 1.3.7 feelpp/spack#2 (#46162)

* package cln 1.3.7 feelpp/spack#2

* add myself as maintainer

* fix style issue, rm blankline

adamjstewart committed:
py-horovod: py-torch 2.1.0 now supported (#46152)

eugeneswalker committed:
e4s ci stacks: add geopm-runtime (#45881)

adamjstewart committed:
py-torchgeo: add v0.6.0 (#46158)

georgiastuart committed:
docs: update conditional definition arch (#46139)

Signed-off-by: Georgia Stuart <gstuart@umass.edu>

Co-authored-by: Jordan Galby <67924449+Jordan474@users.noreply.github.com>

pauleonix committed:
cuda: add v12.6.1 (#46143)

Update build system conflict between CUDA 12.6 and Clang 18

balay committed:
xsdk: remove develop and 0.7.0, and deprecate 0.8.0 (#46121)

balay committed:
petsc, py-petsc4py: add v3.21.5 (#46151)

wdconinc committed:
ddt: deprecate all versions in favor of linaro-forge (#46115)

alalazo commented on var/spack/repos/builtin/packages/seacas/package.py:
```diff
@@ -425,7 +425,7 @@ def cmake_args(self):
         )
 
-        if "+parmetis" in spec:
+        if "+metis" in spec and "+mpi" in spec:
```
```suggestion
        if spec.satisfies("+metis+mpi"):
```

rbberger committed:
kokkos-nvcc-wrapper: add new 4.4.00 version (#46067)

ashermancinelli committed:
[neovim] add utf8proc dependency (#46064)

I believe utf8proc was added as a neovim dependency in neovim/neovim#26165
and is only in the master branch.

wdconinc committed:
py-vector: add v1.4.2, v1.5.0; variant awkward (#46039)

wdconinc committed:
package_base: sort deprecated versions later in `preferred_version` (#46025)

haampie committed:
remove dead code: fs.is_writable_dir was used on file (#46166)

haampie committed:
projections: simplify expression (#46167)

haampie committed:
db: type hints (#46168)

tmadlener committed:
podio: Add the new datasource variant once it is available (#46078)

* podio: Add the new datasource variant once it is available

* Make sure to require a suitable minimal root version

JCGoran committed:
py-sympy: add version 1.13.0 (#46163)

* sympy: add version 1.13.0

* sympy: update deps

omsai committed:
py-cellprofiler: add 4.2.6 new package (#44824)

* py-cellprofiler: add 4.2.6 new package

* py-mysqlclient: Limit pkg-config patch to @1.4:

Co-authored-by: Wouter Deconinck <wdconinc@gmail.com>

---------

Co-authored-by: Wouter Deconinck <wdconinc@gmail.com>

paugier committed:
Add py-pytest-allclose package (#45877)

rbberger committed:
nvpl updates (#45922)

* nvpl-blas: add new version 0.3.0

* nvpl-lapack: add new version 0.2.3.1

haampie committed:
database: remove a few class properties (#46175)

alalazo committed:
solver: minor cleanup and optimization (#46176)

Signed-off-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

stephenswat committed:
root: fix X11 and OpenGL-related issues on macOS (#45632)

* root: Add dependency on libglx

We have been trying to build the Acts package on MacOS, and in this
process we have been running into problems with the ROOT spec on that
operating system; the primary issue we are encountering is that the
compiler is unable to find the `GL/glx.h` header, which is part of glx.
It seems, therefore, that ROOT depends on libglx, but this is not
currently encoded in the spec. This commit ensures that ROOT depends on
the virtual libglx package when both the OpenCL and X11 variants are
enabled.

* Enable builtin glew on MacOS

* Allow `root+opengl+aqua~x` on macOS

aphecetche committed:
root: fix @loader_path on macOS (#44826)

* root: fix @loader_path on macOS

* root: use loader patch from upstream

Co-authored-by: Wouter Deconinck <wdconinc@gmail.com>

* add missing comma

* root: rm unused patch

* root: conflict on macos 12 for @:6.27 when +python

---------

Co-authored-by: Wouter Deconinck <wdconinc@gmail.com>

alalazo committed:
llvm: be more strict with detection (#46179)

msimberg committed:
Add conflict between llvm-amdgpu until version 5 and ninja since version 1.12.0 (#45788)

* Add conflict between llvm-amdgpu and ninja since version 1.12.0

* Update llvm-amdgpu and ninja conflict to extend to 6.0

knoepfel committed:
llvm: improve detection regexes (#46187)

balay committed:
trilinos: update @[master,develop] dependency on kokkos (#46182)

==> Error: InstallError: For Trilinos@[master,develop], ^kokkos version in spec must match version in Trilinos source code. Specify ^kokkos@4.4.00 for trilinos@[master,develop] instead of ^kokkos@4.3.01.

vicentebolea committed:
paraview: add new v5.13.0 release (#46091)

Commit:
Update var/spack/repos/builtin/packages/seacas/package.py

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>
